Consecutive nights

crheidri opened the pull request:
PR for #15. Created an optional --nights input argument to specify the minimum number of contiguous nights to search between dates. Without --nights specified, defaults to original behavior (all nights between --start-date and --end-date).

Example finding spots in Upper Pines, Yosemite this week:
```
$ python camping.py --start-date 2020-03-10 --end-date 2020-03-19 --nights 2 232447
There are campsites available from 2020-03-10 to 2020-03-19!!!
🏕 UPPER PINES (232447): 9 site(s) available out of 240 site(s)

$ python camping.py --start-date 2020-03-10 --end-date 2020-03-19 --nights 3 232447
There are campsites available from 2020-03-10 to 2020-03-19!!!
🏕 UPPER PINES (232447): 1 site(s) available out of 240 site(s)
```

Will warn the user if they give an incorrect number of nights
```
$ python camping.py --start-date 2020-03-15 --end-date 2020-03-17 --nights 5 232447
Setting number of nights to 2.
There are campsites available from 2020-03-15 to 2020-03-17!!!
🏕 UPPER PINES (232447): 2 site(s) available out of 240 site(s)
```

Without specifying `--nights`, defaults to old behavior (all nights between given dates):
```
$ python camping.py --start-date 2020-06-28 --end-date 2020-06-30 234038
There are campsites available from 2020-06-28 to 2020-06-30!!!
🏕 CHISOS BASIN (BIG BEND) (234038): 27 site(s) available out of 62 site(s)
```

This definitely needs review and testing, but so far seems to be working okay on my end. It does require itertools for some list comprehension, unless we figure out a simpler way than my approach :)

banool commented on camping.py:
```diff
@@ -143,6 +143,12 @@ def _main(parks):
     )
+    parser.add_argument(
+        "--nights",
+        default=0,
```
I'd rather there not be a default and just treat the case where this is not set (it is None) as the "all nights" case.

crheidri replied:
set nights=None by default

banool commented on camping.py:
```diff
@@ -143,6 +143,12 @@ def _main(parks):
+    parser.add_argument(
+        "--nights",
+        default=0,
+        help="Number of consecutive nights (default is all nights).",
```
s/default is all nights/default is all nights in the given range/

banool commented on camping.py:
```diff
@@ -62,7 +62,7 @@ def get_name_of_site(park_id):
 
 
-def get_num_available_sites(resp, start_date, end_date):
+def get_num_available_sites(resp, start_date, end_date, nights):
```
Make nights an optional kwarg.

banool commented:
This looks pretty good to me! I've requested a few changes here and there. In your original PR, could you also include test output from when you _don't_ pass in the `nights` flag? 

This is of course in the absence of me having any real tests, so we need to test it manually quite thoroughly 😛.

Thanks for doing this, great feature!

banool commented:
As an additional ask, it'd be great if you could include in the README a little usage sample to point out that this feature exists.

crheidri commented:
Sorry for letting this sit for so long - just pushed some changes addressing your feedback.

- `nights=None` is now an optional kwarg in `get_num_available_sites`
- ensures --nights is a positive integer before calling `_main`
- replaced the `WARN_NIGHTS` global with a call to `LOG.debug`
- edited original PR to show usage without --nights specified
- added example of --nights usage in README
